dGame/dComponents/VendorComponent.cpp

```diff
@@ -1,124 +1,96 @@
+		auto vendorItems = lootTableTable->Query([=](CDLootTable entry) { return (entry.LootTableIndex == lootTableID); });
 		if (lootMatrix.maxToDrop == 0 || lootMatrix.minToDrop == 0) {
-			for (CDLootTable item : vendorItems) {
-				m_Inventory.insert({ item.itemid, item.sortPriority });
+			for (const auto& item : vendorItems) {
+				if (!m_HasStandardCostItems || !m_HasMultiCostItems) {
+					auto itemComponentID = compRegistryTable->GetByIDAndType(item.itemid, eReplicaComponentType::ITEM);
+					auto itemComponent = itemComponentTable->GetItemComponentByID(itemComponentID);
+					if (!m_HasStandardCostItems && itemComponent.baseValue != -1) SetHasStandardCostItems(true);
+					if (!m_HasMultiCostItems && !itemComponent.currencyCosts.empty()) SetHasMultiCostItems(true);
+				}
+				m_Inventory.push_back(SoldItem(item.itemid, item.sortPriority));
 			}
 		} else {
 			auto randomCount = GeneralUtils::GenerateRandomNumber<int32_t>(lootMatrix.minToDrop, lootMatrix.maxToDrop);
 
 			for (size_t i = 0; i < randomCount; i++) {
 				if (vendorItems.empty()) break;
-
 				auto randomItemIndex = GeneralUtils::GenerateRandomNumber<int32_t>(0, vendorItems.size() - 1);
-
-				const auto& randomItem = vendorItems[randomItemIndex];
-
+				const auto& randomItem = vendorItems.at(randomItemIndex);
 				vendorItems.erase(vendorItems.begin() + randomItemIndex);
-
-				m_Inventory.insert({ randomItem.itemid, randomItem.sortPriority });
+				if (!m_HasStandardCostItems || !m_HasMultiCostItems) {
+					auto itemComponentID = compRegistryTable->GetByIDAndType(randomItem.itemid, eReplicaComponentType::ITEM, -1);
+					if (itemComponentID == -1) {
+						Game::logger->Log("VendorComponent", "Attempted to add item %i with ItemComponent ID -1 to vendor %i inventory. Not adding item!", itemComponentID, m_Parent->GetLOT());
+						continue;
+					}
+					auto itemComponent = itemComponentTable->GetItemComponentByID(itemComponentID);
+					if (!m_HasStandardCostItems && itemComponent.baseValue != -1) SetHasStandardCostItems(true);
+					if (!m_HasMultiCostItems && !itemComponent.currencyCosts.empty()) SetHasMultiCostItems(true);
+				}
+				m_Inventory.push_back(SoldItem(randomItem.itemid, randomItem.sortPriority));
 			}
 		}
 	}
 
-	//Because I want a vendor to sell these cameras
-	if (m_Parent->GetLOT() == 13569) {
-		auto randomCamera = GeneralUtils::GenerateRandomNumber<int32_t>(0, 2);
-
-		switch (randomCamera) {
-		case 0:
-			m_Inventory.insert({ 16253, 0 }); //Grungagroid
-			break;
-		case 1:
-			m_Inventory.insert({ 16254, 0 }); //Hipstabrick
-			break;
-		case 2:
-			m_Inventory.insert({ 16204, 0 }); //Megabrixel snapshot
-			break;
-		default:
-			break;
-		}
-	}
-
 	// Callback timer to refresh this inventory.
-	m_Parent->AddCallbackTimer(m_RefreshTimeSeconds, [this]() {
-		RefreshInventory();
-		});
+	if (m_RefreshTimeSeconds > 0.0) {
+		m_Parent->AddCallbackTimer(m_RefreshTimeSeconds, [this]() {
+			RefreshInventory();
+			}
+		);
```

Comment:
I dont think we generally do a new line between the last argument of a function and the parenthesis, but this is a nit.

Reply:
I'll see what the standard is, I do it cause it's easier to see scope top to bottom